OpenUpSA/wazimap-ng
# configure backend the standard OpenUp way

## Deployment

Before deployment

- add ENVIRONMENT_NAME with value from SERVER_INSTANCE

After deployment

- remove DJANGO_CONFIGURATION
- remove SERVER_INSTANCE
- 
Also after deployment, but I can't see these used anywhere anyway

- remove RUN_COLLECTSTATIC
- remove RUN_GUNICORN
- remove RUN_MIGRATIONS
- remove RUN_QCLUSTER

## Description

Configure the app the way we normally configure apps at OpenUp

- sane/safe defaults for production or other new deployments
- app doesn't start if something that must be specific to each environment hasn't been set
- everything is set via an environment variable instead of environment-specific sets of changes
- avoiding blanket settings (using django's DEBUG as little as possible, and used TESTING because it was already there and to limit scope of changes here)
- Django cache is disabled for dev but enabled for tests that are testing it. Can easily be enabled using .env file
- .env is deleted and gitignored to avoid constant diffs or accidentally pushing secrets used in local dev to github

https://12factor.net/config

## Related Issue

https://wazimap.atlassian.net/browse/WNCM-507

## How to test it locally

- check if review app is working as expected
- check if it works as expected in staging
- check it works locally
  - can you start your local instance and use it?
  - can you run tests locally? (see how they're run in the github actions workflow and run that way)

## Changelog

- Remove Django Configurations classes
- switch to a single settings file
- use django-environ to parse environment variables
- document some commonly-modified variables

### Added

### Updated

### Removed


## Checklist

- [ x]  🚀 is the code ready to be merged and go live?
- [ x]  🛠 does it work (build) locally

### Pull Request


### Commits


### Code Quality


### Testing
